### ISSUE TITLE
onConflict: use of VALUES() function in MySQL is deprecated

### ISSUE THREAD

**Issue by sanderua** [knex/knex#4990]
Note from MySQL documentation for INSERT ... ON DUPLICATE KEY UPDATE statement:

> The use of VALUES() to refer to the new row and columns is deprecated beginning with MySQL 8.0.20, and is subject to removal in a future version of MySQL. Instead, use row and column aliases, as described in the next few paragraphs of this section.

https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html

**Comment by mercmobily** [comment]
You’re right: using the MySQL‑specific `VALUES(col)` in `ON DUPLICATE KEY UPDATE` is deprecated in favour of referencing the inserted row via an alias.

This needs work in the MySQL compiler to emit the modern syntax. The fact that the issue is 4 years old makes things even more important. 

Code evidence (current):

- lib/dialects/mysql/query/mysql-querycompiler.js:70 — `_merge(...)` generates `... on duplicate key update col = values(col)`, which triggers the deprecation in MySQL ≥ 8.0.20

Suggested change:

- Compile `INSERT ... AS new ON DUPLICATE KEY UPDATE col = new.col` when building an upsert with `merge`, for MySQL versions where aliasing is supported (8.0.19+). This avoids `VALUES()` and aligns with the documented replacement.
- If version detection is not feasible, consider a configuration flag to opt‑in to alias mode, defaulting to current behaviour for backward compatibility.

I wish I could have said "Let's drop the old syntax" but... Oracle’s policy lists Premier Support through April 2026 and Extended Support through April 2029. So we can talk about it in 2030 I think... 

Until this lands, the practical workaround is to use `knex.raw` for the upsert in MySQL 8.0.20+ or accept the deprecation warning.
